src/web4/handler.rs

```diff
@@ -0,0 +1,331 @@
+use near_sdk::base64::{
+    alphabet,
+    engine::{self, general_purpose},
+    Engine,
+};
+use serde_json::json;
+
+use crate::{
+    web4::types::{Web4Request, Web4Response},
+    Contract, Proposal,
+};
+
+pub const BASE64_ENGINE: engine::GeneralPurpose =
+    engine::GeneralPurpose::new(&alphabet::URL_SAFE, general_purpose::NO_PAD);
+
+pub fn web4_get(contract: &Contract, request: Web4Request) -> Web4Response {
+    let path_parts: Vec<&str> = request.path.split('/').collect();
+
+    let page = path_parts[1];
+    let mut title: String = String::from("near/dev/hub");
+    let mut description: String = String::from("The decentralized home base for NEAR builders");
+    let mut image: String = String::from(
+        "https://i.near.social/magic/large/https://near.social/magic/img/account/devhub.near",
+    );
+    let mut redirect_path: String = String::from("devhub.near/widget/app");
```

Comment:
nit: Explicit type annotation is redundant here since you can see the type on the right.